lib/assertions.js

```diff
@@ -766,39 +766,61 @@ module.exports = (expect) => {
     );
   });
 
-  expect.addAssertion('<array-like> [not] to contain <any+>', function (
-    expect,
-    subject
-  ) {
-    const args = Array.prototype.slice.call(arguments, 2);
-    expect.withError(
-      () => {
-        args.forEach((arg) => {
-          expect(
-            subject &&
-              Array.prototype.some.call(subject, (item) =>
-                expect.equal(item, arg)
-              ),
-            '[not] to be truthy'
-          );
-        });
-      },
-      (e) => {
-        expect.fail({
-          diff:
-            expect.flags.not &&
-            ((output, diff, inspect, equal) =>
-              diff(
-                subject,
-                Array.prototype.filter.call(
-                  subject,
-                  (item) => !args.some((arg) => equal(item, arg))
-                )
-              )),
-        });
+  expect.addAssertion(
+    '<array-like> [not] to [only] contain <any+>',
+    function (expect, subject) {
+      const args = Array.prototype.slice.call(arguments, 2);
+
+      if (expect.flags.only && expect.flags.not) {
+        expect.errorMode = 'bubble';
+        expect.fail(
+          'The "not" flag cannot be used together with "to only contain".'
+        );
```

Comment:
You could avoid this check by passing multiple patterns to `addAssertion` so that the illegal combination is ruled out:

```js
  expect.addAssertion(
    [
      '<array-like> to [only] contain <any+>',
      '<array-like> [not] to contain <any+>',
    ],
```

Reply:
Thanks for the tip :-) That's really neat. I remembered the pattern from earlier work with unexpected, but it's probably just me being too oldschool to really understand the possibilities with the types. Much better solution! :-) 

I'll defer changes until the conversation about the inadequate `expect(subject.length, 'to be', args.length);`-check is resolved though.

Reply:
This is addressed in the latest commit.